test/mocha/30-profile-agents-zcaps-refresh.js

```diff
@@ -0,0 +1,259 @@
+/*!
+ * Copyright (c) 2023 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as bedrock from '@bedrock/core';
+import * as helpers from './helpers.js';
+import * as utils from '@bedrock/profile/lib/utils.js';
+import {profileAgents, profiles} from '@bedrock/profile';
+import {EdvClient} from '@digitalbazaar/edv-client';
+import {getAppIdentity} from '@bedrock/app-identity';
+import {httpsAgent} from '@bedrock/https-agent';
+import {keyResolver} from '@bedrock/profile/lib/keyResolver.js';
+import {klona} from 'klona';
+import {KmsClient} from '@digitalbazaar/webkms-client';
+import {mockData} from './mock.data.js';
+import {v4 as uuid} from 'uuid';
+
+const {
+  getEdvConfig,
+  getEdvDocument,
+  getProfileSigner,
+} = utils;
+
+describe('Refresh Profile Agent Zcaps', () => {
+  let edvOptions;
+  let keystoreOptions;
+  // mock session authentication for delegations endpoint
+  let passportStub;
+  before(async () => {
+    await helpers.prepareDatabase(mockData);
+    passportStub = helpers.stubPassport();
+  });
+  beforeEach(async () => {
+    // top-level applications must create meters
+    const {keys} = getAppIdentity();
+    const invocationSigner = keys.capabilityInvocationKey.signer();
+
+    const {id: edvMeterId} = await helpers.createMeter({type: 'edv'});
+    const {id: kmsMeterId} = await helpers.createMeter({type: 'webkms'});
+    edvOptions = {
+      profile: {
+        baseUrl: bedrock.config.server.baseUri,
+        meterId: edvMeterId,
+        meterCapabilityInvocationSigner: invocationSigner
+      }
+    };
+    keystoreOptions = {
+      profileAgent: {
+        meterId: kmsMeterId,
+        meterCapabilityInvocationSigner: invocationSigner
+      },
+      profile: {
+        meterId: kmsMeterId,
+        meterCapabilityInvocationSigner: invocationSigner
+      }
+    };
+  });
+  after(() => {
+    passportStub.restore();
+  });
+
+  describe('profileAgents.getAll() API', () => {
+    it('should refresh profile agent user doc zcaps and profile agent zcaps ' +
```

Comment:
Need an additional test that waits for the refresh threshold and then kicks off ~10 calls to get all profile agents concurrently and ensures that just one succeeds at performing the refresh while the other 9 do not fail to return the properly refreshed records (and ensure the EDV doc sequence is only incremented once).

Reply:
@dlongley I'm unclear how this should be tested. I've added the retry until the update succeeds code here [40cfdd3](https://github.com/digitalbazaar/bedrock-profile/pull/99/commits/40cfdd38b32c25a9c9b383f5096fdb932160878b). Can this be turned into an issue and addressed some other time?

Reply:
@dlongley Actually, I think I got it now, I've fixed the retry logic here https://github.com/digitalbazaar/bedrock-profile/pull/101 and added test. Let me know if that looks ok to you.